edition/voxel_tool.cpp

```diff
@@ -189,7 +189,33 @@ void VoxelTool::do_sphere(Vector3 center, float radius) {
 }
 
 void VoxelTool::do_box(Vector3i begin, Vector3i end) {
-	ERR_PRINT("Not implemented");
+	Vector3i::sort_min_max(begin, end);
+	Rect3i box = Rect3i::from_min_max(begin, end + Vector3i(1,1,1));
+
+	if (!is_area_editable(box)) {
+		PRINT_VERBOSE("Area not editable");
+		return;
+	}
+
+	if (_channel == VoxelBuffer::CHANNEL_SDF) {
+
+		Vector3 center = (box.pos + (box.size/2)).to_vec3();
+
+		box.for_each_cell([this, center](Vector3i pos) {
+			float d = pos.to_vec3().distance_to(center);
+			_set_voxel_f(pos, sdf_blend(d, get_voxel_f(pos), _mode));
```

Comment:
Distance to center is not box SDF, it is sphere SDF. Your previous code was good, it just missed the use of `sdf_blend` instead of `_set_voxel_f`. Here instead of distance you may want to send either -1 or 1 across the whole box like you did before.

You may also want to test your code with the demo project.